pkg/webhook/resources/upgrade/validator.go

```diff
@@ -65,5 +85,58 @@ func (v *upgradeValidator) Create(request *types.Request, newObj runtime.Object)
 		return werror.NewConflict(msg)
 	}
 
+	hasDegradedVolume, err := v.hasDegradedVolume()
+	if err != nil {
+		return werror.NewInternalError(err.Error())
+	}
+
+	if hasDegradedVolume {
+		return werror.NewBadRequest("there are degraded volumes, please check all volumes are healthy")
```

Comment:
Note: we will need to consider exposing the LH volume status in future versions; otherwise it would hard for users to tell and fix those degraded volumes.

Reply:
Create an issue to track this; thanks! https://github.com/harvester/harvester/issues/2856